### PR TITLE
ROL using inline assembly

### PR DESCRIPTION
**Pull request by raven02**

### CODE REVIEW

**Comment by Nekotekina** [comment]
MSVC doesn't support inline asm, need to keep two implementations with ifdef.

**Comment by raven02** [comment]
which #ifdef should use in the case to embrace these rotate functions ?

**Comment by kd-11** [comment]
@raven02 #ifdef _MSC_VER

**Comment by raven02** [comment]
@Nekotekina / @kd-11 looks correct ?

**Comment by danilaml** [comment]
@Nekotekina @raven02 can't it be done with intrinsics?
Also, it'd be nice to check if compiler doesn't generate those instructions already.

**Comment by danilaml** [comment]
I did a quick check, and it seems that at least GCC translates it to `rol` just fine.

**Comment by kd-11** [comment]
The exclusion block should be fine, although travis-ci doesn't like the asm block for some reason

**Comment by danilaml** [comment]
Btw, there is non asm/intrinsic solution, that should compile to simple `rol` on `icc`, `gcc` and `clang` (not sure about MSVC).
```C++
inline u64 rol64(u64 x, u64 n) // same for other types
{
  const auto mask = 8*sizeof(x) - 1;
  n &= mask;  // avoid UB
  return ( x << n ) | ( x >> ((-n)&mask) );
}
```

**Comment by Nekotekina** [comment]
I think rol32 and rol64 are already fine. The problem is that only rol8 and rol16 are not well compiled.

File: rpcs3/Emu/Cell/PPUInterpreter.cpp [hunk 1]
```diff
@@ -24,10 +26,6 @@ inline s64 MULH64(s64 a, s64 b)
 #endif
 
 #if defined(_MSC_VER)
-#define rol8 _rotl8
-#define rol16 _rotl16
```
Comment (danilaml) [inline rpcs3/Emu/Cell/PPUInterpreter.cpp]: You've removed rol8/16 from MSVC.

**Comment by raven02** [comment]
I'll rework it .

**Pull request closed without merging**